src/App.tsx

```diff
@@ -14,11 +14,117 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  selectedGoods: string[];
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  getMessage = () => {
+    let message = '';
+    const { selectedGoods } = this.state;
+
+    switch (selectedGoods.length) {
+      case 0:
+        message = 'No goods selected';
+        break;
+
+      case 1:
+        message = `${selectedGoods[0]} is selected`;
+        break;
+
+      default:
+        message = `${selectedGoods.slice(0, selectedGoods.length - 1).join(', ')} and ${selectedGoods[selectedGoods.length - 1]} are selected`;
+        break;
+    }
```

Comment:
reassigning of variables - not very good practice

```suggestion
    const { selectedGoods } = this.state;

    switch (selectedGoods.length) {
      case 0:
        return 'No goods selected';

      case 1:
        return `${selectedGoods[0]} is selected`;

      default:
        return`${selectedGoods.slice(0, selectedGoods.length - 1).join(', ')} and ${selectedGoods[selectedGoods.length - 1]} are selected`;
    }
```